lib/Basic/FileSystem.cpp

```diff
@@ -62,6 +62,19 @@ class LocalFileSystem : public FileSystem {
     }
     return std::unique_ptr<llvm::MemoryBuffer>(result->release());
   }
+
+  virtual bool remove(const std::string& path) override {
+    struct stat statbuf;
+    if (sys::stat(path.c_str(), &statbuf) != 0) {
```

Comment:
In general, it is usually better to simply try the operation, then report the failure if it doesn't succeed (by checking the errno result). This uses half as many system calls.

In this case, what probably makes sense is to first try the sys::unlink(), and then if that fails because the thing is a directory, then do the rmdir (or `rm -r`, once supported).